### PR TITLE
[1/N] Replaces user reader and writer with repo

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR replaces the current user reader and writer usage with the new repo/model approach.

## Related issue number (if any)
ENG 2046

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/workflow/utils/database.go [hunk 1]
```diff
@@ -395,35 +362,102 @@ func CreateWorkflowDagResult(
...
+			workflow.Name,
+		)
+	}
+
+	notificationAssociation := notification.NotificationAssociation{
+		Object: notification.WorkflowDagResultObject,
+		Id:     dagResult.ID,
 	}
+
+	// TODO: Create notification for all watchers
```
Comment (eunice-chan) [inline src/golang/lib/workflow/utils/database.go:451]: Should we create a task in Linear for this TODO?

File: src/golang/lib/context/context.go [hunk 2]
```diff
@@ -76,10 +76,10 @@ func ParseAqContext(ctx context.Context) (*AqContext, int, error) {
...
-		User: user.User{
-			Id:             userId,
-			OrganizationId: organizationId,
-			Auth0Id:        auth0Id,
+		User: models.User{
+			ID:      userId,
+			OrgID:   organizationId,
+			Auth0ID: auth0Id,
 		},
 		RequestId:     requestId,
```
Comment (eunice-chan) [inline src/golang/lib/context/context.go:84]: RequestId -> RequestID

File: src/golang/lib/context/context.go [hunk 3]
```diff
@@ -76,10 +76,10 @@ func ParseAqContext(ctx context.Context) (*AqContext, int, error) {
 	storageConfig := config.Storage()
 
 	return &AqContext{
-		User: user.User{
-			Id:             userId,
-			OrganizationId: organizationId,
-			Auth0Id:        auth0Id,
+		User: models.User{
```
Comment (eunice-chan) [inline src/golang/lib/context/context.go:79]: Update "Id"s to "ID"s. Perhaps we can do that in a different PR.

File: src/golang/cmd/executor/executor/workflow.go [hunk 4]
```diff
@@ -38,8 +38,9 @@ func NewWorkflowExecutor(spec *job.WorkflowSpec, base *BaseExecutor) (*WorkflowE
 		return nil, err
 	}
 
-	engineReaders := GetEngineReaders(base.Readers)
-	engineWriters := GetEngineWriters(base.Writers)
+	engineReaders := getEngineReaders(base.Readers)
```
Comment (eunice-chan) [inline src/golang/cmd/executor/executor/workflow.go:41]: To confirm: will`getEngineRepos` be replacing `getEngineReaders` and `getEngineWriters` once it is completely merged in?

**Review by eunice-chan — APPROVED** [review]

File: src/golang/cmd/executor/executor/workflow.go (hunk 4 above)
Comment (saurav-c) [reply src/golang/cmd/executor/executor/workflow.go:41]: Yup thats the plan

File: src/golang/lib/context/context.go (hunk 3 above)
Comment (saurav-c) [reply src/golang/lib/context/context.go:79]: Yup we can do as much as possible in these PRs but here is the task for a catch all to do a post cleanup https://linear.app/aqueducthq/issue/ENG-2077/clean-up-codebase-naming-conventions-for-id-db-dag-etc

File: src/golang/lib/workflow/utils/database.go (hunk 1 above)
Comment (saurav-c) [reply src/golang/lib/workflow/utils/database.go:451]: https://linear.app/aqueducthq/issue/ENG-2079/remove-assumption-of-1-watcher-for-each-workflow-when-creating
